src/guiguts/tools/levenshtein.py

```diff
@@ -0,0 +1,683 @@
+            rerun_command=lambda: levenshtein_check(project_dict),
+        )
+        frame = ttk.Frame(checker_dialog.header_frame)
+        frame.grid(column=0, row=1, columnspan=2, sticky="NSEW")
+        ttk.Label(
+            frame,
+            text="Edit Distance:",
+        ).grid(row=0, column=1, sticky="NSE", padx=(0, 5))
+        ttk.Radiobutton(
+            frame,
+            text="1",
+            variable=self.edit_distance,
+            value=LevenshteinEditDistance.ONE,
+            takefocus=False,
+        ).grid(row=0, column=2, sticky="NSE", padx=2)
+        ttk.Radiobutton(
+            frame,
+            text="2",
+            variable=self.edit_distance,
+            value=LevenshteinEditDistance.TWO,
+            takefocus=False,
+        ).grid(row=0, column=3, sticky="NSE", padx=2)
+
+        ToolTip(
+            checker_dialog.text,
+            "\n".join(
+                [
+                    "Left click: Select & find highlighted word in file",
+                    "Right click: Remove line with highlighted word from list",
+                    "Shift Right click: Also remove all matching lines from list",
+                ]
+            ),
+            use_pointer_pos=True,
+        )
+
+        checker_dialog.reset()
+
+        # Convert to int the edit distance used last time Levenshtein was run or default if first run.
+        distance_to_check = 1
+        if self.edit_distance.get() == "two":
+            distance_to_check = 2
+
+        # Build maps of the good words and the suspect words in the file.
+        # Ideally should only do this once and not every time the tool is
+        # rerun.
+        map_words_in_file(project_dict)
+
+        # For each suspect word, check against each good word in turn.
+        # Very compute intensive. Where 90% of the execution time is consumed.
+        distance_check_words()
+
+        # Did distance_check_words() find any word pairs that met
+        # the required Levenshtein criteria?
+        if len(distance_check_results) == 0:
+            # No.
+            checker_dialog.add_entry("No Levenshtein edit distance queries reported")
+            checker_dialog.add_entry("")
+        else:
+            # Yes.
+            for result_tuple in distance_check_results:
+                unpack_and_write_to_dialog(result_tuple)
+
+        # Calculate and display execution time of run.
+        prog_end = time.time()
+        checker_dialog.add_footer(
+            f"Execution time: {(prog_end - prog_start):.2f} seconds"
+        )
+
+        checker_dialog.display_entries()
+
+
+def levenshtein_check(project_dict: ProjectDict) -> None:
+    """Do Levenshtein edit distance checks"""
+
+    global _the_levenshtein_checker
+
+    if not tool_save():
+        return
+
+    if _the_levenshtein_checker is None:
+        try:
+            _the_levenshtein_checker = LevenshteinChecker()
+        except DictionaryNotFoundError as exc:
```

Comment:
When the `LevenshteinChecker` object is created here, it doesn't do any spelling, so can never raise the `DictionaryNotFoundError` exception, so there's not point having the try/except.
However, during running, where the Spellcheck object is created, the exception can be raised. See comment there for more details.